smithy-model/src/main/java/software/amazon/smithy/model/shapes/ResourceShape.java

```diff
@@ -145,6 +148,14 @@ public boolean hasIdentifiers() {
         return !identifiers.isEmpty();
     }
 
+    public Map<String, ShapeId> getProperties() {
+        return properties;
+    }
+
+    public boolean hasProperties() {
```

Comment:
is this necessary?

Reply:
Not yet. I'm anticipating usage in another section of the code. `hasIdentifiers()` is called from ModelSerializer.java and it seems almost sure this will be needed in the same place.